src/source/Signaling/LwsApiCalls.c

```diff
@@ -1718,8 +1751,11 @@ STATUS terminateConnectionWithStatus(PSignalingClient pSignalingClient, SERVICE_
         ATOMIC_STORE_BOOL(&pSignalingClient->pOngoingCallInfo->cancelService, TRUE);
     }
 
-    // Wake up the service event loop
-    CHK_STATUS(wakeLwsServiceEventLoop(pSignalingClient));
+    // Wake up the service event loop for all of the protocols
+    for (i = 0; i < LWS_PROTOCOL_COUNT; i++) {
```

Comment:
Since we removed the locks that are in `wakeLwsServiceEventLoop`, shouldn't we need to acquire a lock here?

Reply:
Good question. The wake up calls lws_callback_on_writable which is pretty much the only API that is re-entrant. The question then is whether the earlier check for if (pSignalingClient->currentWsi[protocolIndex] != NULL) stays valid. I traced the calls and it seems that the higher level state machine locks prevent this case.